src/argilla_server/models/database.py

```diff
@@ -241,7 +242,7 @@ class Question(DatabaseModel):
 
     __table_args__ = (UniqueConstraint("name", "dataset_id", name="question_name_dataset_id_uq"),)
 
-    @property
+    @cached_property
```

Comment:
Are we sure that we will not have problems with this like updating the settings somehow for a `Question` instance and after that using again `parsed_settings` but it will not be updated because it was cached with previous parsed settings?

I usually try to avoid [memoizations](https://en.wikipedia.org/wiki/Memoization) for the previous reason.